processing/src/main/java/org/apache/druid/segment/filter/RegexFilter.java

```diff
@@ -100,7 +100,7 @@ public boolean equals(Object o)
         return false;
       }
       PatternDruidPredicateFactory that = (PatternDruidPredicateFactory) o;
-      return Objects.equals(pattern, that.pattern);
+      return Objects.equals(pattern.toString(), that.pattern.toString());
```

Comment:
I think the hashCode should use the toString as well

Reply:
🤦